java/sage/MiniClientSageRenderer.java

```diff
@@ -1328,6 +1328,29 @@ private boolean checkForAudioOutputChangeUpdates()
     return true;
   }
 
+  private boolean checkForMenuHintUpdates()
+  {
+    if (menuHint.hasHint())
+    {
+      try
+      {
+        int propRV = sendSetProperty("MENU_HINT", menuHint.format());
```

Comment:
I'd suggest getting the format() first, and storing it in a local var and clearing the hint all in a block synchronized around the menuHint object. Then also in setMenuHint, synchronize the code in there around the menuHint object as well. I realize the other code isn't done this way...but it probably should have been. :)